src/azure-cli/azure/cli/command_modules/sql/_params.py

```diff
@@ -1353,20 +1386,47 @@ def _configure_security_policy_storage_params(arg_ctx):
             ])
 
         c.argument('administrator_login',
-                   required=True)
+                   required=False)
 
         c.argument('administrator_login_password',
-                   required=True)
+                   required=False)
 
         c.argument('assign_identity',
                    options_list=['--assign-identity', '-i'],
                    help='Generate and assign an Azure Active Directory Identity for this server '
                    'for use with key management services like Azure KeyVault.')
 
+        c.argument('enable_ad_only_auth',
+                   options_list=['--enable-ad-only-auth'],
+                   help='Enable Azure Active Directory Only Authentication for this server.')
+
+        c.argument('external_admin_name',
+                   options_list=['--external-admin-name'],
+                   help='Display name of the Azure AD administrator user, group or application.')
+
+        c.argument('external_admin_sid',
+                   options_list=['--external-admin-sid'],
+                   help='The unique ID of the Azure AD administrator. Object Id for User or Group, '
+                   'Client Id for Applications')
+
+        c.argument('external_admin_principal_type',
+                   options_list=['--external-admin-principal-type'],
+                   help='User, Group or Application')
+
     with self.argument_context('sql server update') as c:
         c.argument('administrator_login_password',
                    help='The administrator login password.')
 
+    with self.argument_context('sql server show') as c:
+        c.argument('expand_ad_admin',
```

Comment:
does it accept any value?

Reply:
it's a boolean, and is used as only an argument -expand-ad-admin

Reply:
If it is boolean, please use get_three_state_flags()

Reply:
okay sure

Reply:
if this flag is present we expand, we don't support false here